karapace/typing.py

```diff
@@ -59,23 +59,55 @@ class Mode(StrEnum):
     readwrite = "READWRITE"
 
 
-class Version(int):
+class Version:
     LATEST_VERSION_TAG: ClassVar[str] = "latest"
     MINUS_1_VERSION_TAG: ClassVar[int] = -1
 
-    def __new__(cls, version: int) -> Version:
+    def __init__(self, version: int) -> None:
         if not isinstance(version, int):
             raise InvalidVersion(f"Invalid version {version}")
-        if (version < cls.MINUS_1_VERSION_TAG) or (version == 0):
+        if (version < Version.MINUS_1_VERSION_TAG) or (version == 0):
             raise InvalidVersion(f"Invalid version {version}")
-        return super().__new__(cls, version)
+        self._value = version
 
     def __str__(self) -> str:
-        return f"{int(self)}"
+        return f"{int(self._value)}"
 
     def __repr__(self) -> str:
-        return f"Version={int(self)}"
+        return f"Version({int(self._value)})"
+
+    def __gt__(self, other: Version) -> bool:
+        if isinstance(other, Version):
+            return self._value > other.value
+        raise NotImplemented(f">Cannot compare {repr(other)} with {repr(self)}")
```

Comment:
issue: `NotImplemented` is not an exception, and it should not be raised, it should be returned.

Reply:
It also isn't callable, these lines should be just simply `return NotImplemented`.

```pycon
>>> NotImplemented("Foo")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NotImplementedType' object is not callable
```